arangod/Sharding/ShardingInfo.cpp

```diff
@@ -307,8 +308,11 @@ void ShardingInfo::toVelocyPack(VPackBuilder& result, bool translateCids) const
   result.add(StaticStrings::WriteConcern, VPackValue(_writeConcern));
   result.add(StaticStrings::MinReplicationFactor, VPackValue(_writeConcern));
 
-  if (!_distributeShardsLike.empty() && ServerState::instance()->isCoordinator()) {
-    if (translateCids) {
+  if ((!_distributeShardsLike.empty() && ServerState::instance()->isCoordinator()) ||
+      (!_distributeShardsLike.empty() && ServerState::instance()->isSingleServer() && isSmartOrSatellite)) {
+    // We either want to expose _distributeShardsLike if we're either on a Coordinator
+    // Or we have found a Smart or Satellite collection on a single server instance.
+    if (translateCids && !ServerState::instance()->isSingleServer()) {
```

Comment:
Why not on SingleServer?
In theory we can have the same layout here and have a number in distributeShardsLike or not?
Or is the number a relict from the past?

Reply:
Yes it is enough. But I agree - I'll restructure this a bit to make it more readable. Commit comes soon.

Reply:
```
commit 39d33396b2eb7044c9f05d08b06cce17703ba4c2 (HEAD -> feature/single-server-smart-graph, origin/feature/single-server-smart-graph)
Author: Heiko Kernbach <heiko@arangodb.com>
Date:   Mon Nov 15 18:43:33 2021 +0100

    make code more readable
    ```

Reply:
```
+  if (!_distributeShardsLike.empty()) {
+    if (ServerState::instance()->isCoordinator()) {
+      // We either want to expose _distributeShardsLike if we're either on a Coordinator
+      if (translateCids) {
+        CollectionNameResolver resolver(_collection->vocbase());
+
+        result.add(StaticStrings::DistributeShardsLike,
+                   VPackValue(resolver.getCollectionNameCluster(DataSourceId{
+                       basics::StringUtils::uint64(distributeShardsLike())})));
+      } else {
+        result.add(StaticStrings::DistributeShardsLike,
+                   VPackValue(distributeShardsLike()));
+      }
+    } else if (ServerState::instance()->isSingleServer()) {
+      // Or we have found a Smart or Satellite collection on a single server instance.
       result.add(StaticStrings::DistributeShardsLike, VPackValue(distributeShardsLike()));
     }
   }
```